lib/storage/src/content_manager/consensus_manager.rs

```diff
@@ -600,10 +600,27 @@ impl<C: CollectionContainer> ConsensusManager<C> {
         consensus_tick: Duration,
         timeout: Duration,
     ) -> Result<(), ()> {
-        let start = Instant::now();
+        match tokio::time::timeout(
+            timeout,
+            self.inner_wait_for_consensus_commit(commit, term, consensus_tick),
+        )
+        .await
+        {
+            Ok(_) => Ok(()),
+            Err(_) => Err(()),
+        }
+    }
+
+    async fn inner_wait_for_consensus_commit(
+        &self,
+        commit: u64,
+        term: u64,
+        consensus_tick: Duration,
+    ) -> Result<(), ()> {
+        let mut interval = tokio::time::interval(consensus_tick);
 
-        // TODO: naive approach with spinlock for waiting on commit/term, find better way
-        while start.elapsed() < timeout {
+        loop {
+            interval.tick().await;
```

Comment:
We don't want to wait on the first run.

Reply:
The first tick is triggered immediately as per docs (https://docs.rs/tokio/latest/tokio/time/struct.Interval.html).